src/cameraGeom/Amplifier.cc

```diff
@@ -0,0 +1,306 @@
+                                   std::string const& unit) {
+        AmpInfoBoxKey result;
+        result._min = table::Point2IKey::addFields(
+            schema,
+            schema.join(name, "min"),
+            doc + ", min point",
+            unit
+        );
+        result._dimensions = table::Point2IKey::addFields(
+            schema,
+            schema.join(name, "extent"),
+            doc + ", extent",
+            unit
+        );
+        return result;
+    }
+
+    // Default constructor; instance will not be usable unless subsequently assigned to.
+    AmpInfoBoxKey() noexcept = default;
+
+    /*
+     *  Construct from a subschema, assuming _min_x, _max_x, _min_y, _max_y subfields
+     *
+     *  If a schema has "a_min_x" and "a_min_x" (etc) fields, this constructor allows you to construct
+     *  a BoxKey via:
+     *
+     *      BoxKey<Box> k(schema["a"]);
+     */
+    AmpInfoBoxKey(table::SubSchema const& s) : _min(s["min"]), _dimensions(s["extent"]) {}
+
+    AmpInfoBoxKey(AmpInfoBoxKey const&) noexcept = default;
+    AmpInfoBoxKey(AmpInfoBoxKey&&) noexcept = default;
+    AmpInfoBoxKey& operator=(AmpInfoBoxKey const&) noexcept = default;
+    AmpInfoBoxKey& operator=(AmpInfoBoxKey&&) noexcept = default;
+    ~AmpInfoBoxKey() noexcept override = default;
+
+    // Get a Box from the given record
+    lsst::geom::Box2I get(table::BaseRecord const& record) const override {
+        return lsst::geom::Box2I(record.get(_min),
+                                 lsst::geom::Extent2I(record.get(_dimensions)));
+    }
+
+    // Set a Box in the given record
+    void set(table::BaseRecord& record, lsst::geom::Box2I const& value) const override {
+        record.set(_min, value.getMin());
+        record.set(_dimensions, lsst::geom::Point2I(value.getDimensions()));
+    }
+
+    // Return True if both the min and max PointKeys are valid.
+    bool isValid() const noexcept { return _min.isValid() && _dimensions.isValid(); }
+
+private:
+    table::Point2IKey _min;
+    table::Point2IKey _dimensions;
+};
+
+
+struct RecordSchemaHelper {
+    table::Schema schema;
+    table::Key<std::string> name;
+    AmpInfoBoxKey bbox;
+    table::Key<double> gain;
+    table::Key<double> saturation;
+    table::Key<double> suspectLevel;
+    table::Key<double> readNoise;
+    table::Key<int> readoutCorner;
+    table::Key<table::Array<double> > linearityCoeffs;
+    table::Key<std::string> linearityType;
+    table::Key<table::Flag> hasRawInfo;
+    AmpInfoBoxKey rawBBox;
+    AmpInfoBoxKey rawDataBBox;
+    table::Key<table::Flag> rawFlipX;
+    table::Key<table::Flag> rawFlipY;
+    table::PointKey<int> rawXYOffset;
+    AmpInfoBoxKey rawHorizontalOverscanBBox;
+    AmpInfoBoxKey rawVerticalOverscanBBox;
+    AmpInfoBoxKey rawPrescanBBox;
+    table::Key<double> linearityThreshold;
+    table::Key<double> linearityMaximum;
+    table::Key<std::string> linearityUnits;
```

Comment:
I think it could be solved without needing to use reflection, but I'll grant that if you can't do a simple refactor then it's best left to another ticket.